### PR TITLE
View All Case Details

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
Offers user ability to view all case and event details of selected case

# What has changed
New page to view all case details
New templates
Tests

# How to test?
Build case-api image and run ops-ui locally. Select a case and check you can see the details
Load sample file locally and go to http://localhost:8234/

# Links
https://trello.com/c/ZxxGq4aT/1222-r-ops-ui-view-all-case-details-13
https://rops-prototypes.netlify.app/prototypes/rops-management/version-2/1-acacia-avenue/index.html

### CODE REVIEW

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Screenshot 2020-08-27 at 15 12 38

Can you pretty-print the JSON payload a bit to make it more readable?

File: census_rm_ops_ui/templates/view_case_details.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{% extends "layouts/base.html" %}
```
Comment (AdamHawtin) [inline census_rm_ops_ui/templates/view_case_details.html:1]: We've used 2 spaces indents on the other HTML templates, would be nice to format this one to match 💅

File: census_rm_ops_ui/templates/view_case_details.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            </p>
+    <h1>Case reference {{ case_details.caseRef }}</h1>
+
+    {%  for key, value in case_details.items() %}
+        <p><b>{{ key }}</b> : {{ value }}</p>
+    {% endfor %}
+    <h1>Event history</h1>
+    {% for event in case_events %}
+        {%  for key, value in event.items() %}
+            <p>{{ key }} : {{ value }}</p>
```
Comment (AdamHawtin) [inline census_rm_ops_ui/templates/view_case_details.html]: Could make this key bold the same as above. Normally leave a space after the colon not before too.

File: census_rm_ops_ui/views/case_details.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+@case_details_bp.route('/')
+def case_details_results():
+    case_id = request.args.get('case_id')
+
+    case_details = case_controller.get_all_case_details(case_id, current_app.config['CASE_API_URL'])
+    case_events = case_details['events']
+    sorted_case_events = sorted(case_events, key=lambda case_event: case_event['eventDate'])
+    case_details.pop('events')
```
Comment (AdamHawtin) [inline census_rm_ops_ui/views/case_details.py]: `pop` returns the element it pops
```suggestion
    case_events = case_details.pop('events')
    sorted_case_events = sorted(case_events, key=lambda case_event: case_event['eventDate'])
```

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
Couple of small bits

**Review by NickGrantX — APPROVED** [review]
Very nice

**Review by AdamHawtin — APPROVED** [review]
👍 Perfectly functional first iteration, hoping we can get some UI help on this ASAP to turn it into a slightly more user-friendly page.

I think all it really needs is a bit of thought on the layout and to make the events section/details collapsible.

**Pull request merged**